parsers/ENTSOE.py

```diff
@@ -723,52 +724,28 @@ def parse_scalar(
     return list(zip(values, datetimes, strict=True))
 
 
-def create_production_storage(
-    fuel_code: str, quantity: float, logger: Logger, zoneKey: ZoneKey
-) -> tuple[ProductionMix | None, StorageMix | None]:
-    production = ProductionMix()
-    storage = StorageMix()
-    fuel_em_type = ENTSOE_PARAMETER_BY_GROUP[fuel_code]
-    if fuel_code in ENTSOE_STORAGE_PARAMETERS:
-        # Only include consumption if it's for storage. In other cases
-        # it is power plant self-consumption which should be ignored.
-        storage.add_value(fuel_em_type, -quantity)
-        return None, storage
-    if 0 > quantity > -50:
-        logger.info(
-            f"Self consumption value {quantity} for {fuel_em_type} has been set to 0.",
-            extra={"key": zoneKey, "fuel_type": fuel_em_type},
-        )
-        quantity = 0
```

Comment:
The tests needed changing because this was removed and is now using the internal correction in the parser methods.

Reply:
We have now abstracted that away into the parser classes so it's consistent across the parsers and logs the corrected modes as a part of the returned data. So this custom check is no longer necessary.